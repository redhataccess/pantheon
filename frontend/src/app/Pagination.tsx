import React, { Fragment } from 'react';
import {Button,Badge,Level, LevelItem} from '@patternfly/react-core';

export interface IProps{
    handleMoveLeft: Function
    handleMoveRight: Function
    pageNumber : number
    nextPageRecordCount: number
    noOfRecordsOnPage: number
}

export class Pagination extends React.Component<IProps> {

private fetchBlocks = () => {
    const totalBlocks = 2;
    const PREVIOUS_PAGE = 'PREVIOUS';
    const NEXT_PAGE = 'NEXT';
    const pages = [PREVIOUS_PAGE,NEXT_PAGE,totalBlocks];
    return pages;
}

render() {

    const blocks = this.fetchBlocks();
    return (
      <Fragment>
        <Level gutter="md">
          <LevelItem>
            <nav aria-label="Countries Pagination">
              {blocks.map((page, index) => {
                if (page === "PREVIOUS") return (
                  this.props.pageNumber === 1 ?
                    <Button isDisabled href="#" target="_blank" variant="primary" onClick={() => this.props.handleMoveLeft()}>Previous</Button>
                    :
                    <Button href="#" variant="primary" onClick={() => this.props.handleMoveLeft()}>Previous</Button>
                );

                if (page === "NEXT") return (
                  this.props.nextPageRecordCount === 0 ?
                    <Button isDisabled href="#" target="_blank" variant="secondary" onClick={() => this.props.handleMoveRight()}>Next</Button>
                    :
                    <Button href="#" variant="secondary" onClick={() => this.props.handleMoveRight()}>Next</Button>
                );
                
                return (
                  <Badge>Page No: {this.props.pageNumber}</Badge>
                );
              })}
            </nav>
          </LevelItem>
          <LevelItem>
            <Badge>No. of records on page: {this.props.noOfRecordsOnPage}</Badge>
          </LevelItem>
        </Level>
      </Fragment>
    );

  }
} 

export default IProps;