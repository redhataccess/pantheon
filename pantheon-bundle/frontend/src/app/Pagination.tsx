import React, { Fragment } from "react";
import { DropdownItem, Level, LevelItem } from "@patternfly/react-core";
import { ContextSelector, ContextSelectorItem } from "@patternfly/react-core";
import { Dropdown } from "@app/Dropdown";

export interface IProps {
  handleMoveLeft: () => any
  handleMoveRight: () => any
  handleMoveToFirst: () => any
  handlePerPageLimit: (itemsPerPage) => any
  handleItemsPerPage: (itemsPerPage) => any
  pageNumber: number
  nextPageRecordCount: number
  perPageLimit: number
  showDropdownOptions: boolean
  bottom: boolean
  className?: string
  currentBulkOperation: string
}

class Pagination extends React.Component<IProps> {

  public dropdownItems = [
    "5 items per page",
    "25 items per page",
    "50 items per page",
    "100 items per page"
  ];

  public state = {
    filteredItems: this.dropdownItems,
    isExpanded: false,
    isOpen: false,
    renderSearch: false,
    searchValue: "",
    selected: this.dropdownItems[0]

  };

  public firstPageButton = true;
  public previousPageButton = true;
  public nextPageButton = true;

  public render() {
    const { isOpen, filteredItems } = this.state;

    if (this.props.pageNumber === 1 && this.props.nextPageRecordCount !== 0) {
      this.firstPageButton = true;
      this.previousPageButton = true;
      this.nextPageButton = false;
    }
    if (this.props.nextPageRecordCount === 0 && this.props.pageNumber !== 1) {
      this.firstPageButton = false;
      this.previousPageButton = false;
      this.nextPageButton = true;
    }
    if (this.props.pageNumber !== 1 && this.props.nextPageRecordCount !== 0) {
      this.firstPageButton = false;
      this.previousPageButton = false;
      this.nextPageButton = false;
    }
    if (this.props.pageNumber === 1 && this.props.nextPageRecordCount === 0) {
      this.firstPageButton = true;
      this.previousPageButton = true;
      this.nextPageButton = true;
    }
    if(this.props.currentBulkOperation !== ""){
      this.firstPageButton = true;
      this.previousPageButton = true;
      this.nextPageButton = true;
    }

    return (
      <Fragment>
        <Level>
          <LevelItem />
          <LevelItem />
          <LevelItem>
            <nav aria-label="Pagination" className={this.props.className}>
              <div>
                <div>
                  <div className="example">
                    <div className="ws-preview">
                      <div className="pf-c-pagination" id="pagination-options-menu-top">
                        <div className="pf-c-dropdown">
                          <div className="pf-c-options-menu__toggle pf-m-plain pf-m-text">
                            {!this.props.showDropdownOptions && !this.props.bottom &&
                              <ContextSelector
                                isOpen={isOpen}
                                onToggle={this.onToggle}
                              />}
                            {this.props.showDropdownOptions &&
                              <Dropdown
                                perPageValue={this.dropDownValue}
                                newPerPagevalue={this.props.perPageLimit + " items per page"}
                              />
                            }
                            <button disabled={this.firstPageButton} data-action="first" aria-label="Go to first page" className="pf-c-button pf-m-plain" type="button" onClick={this.props.handleMoveToFirst}><svg fill="currentColor" height="1em" width="1em" viewBox="0 0 448 512" aria-hidden="true" role="img">
                              <path d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z" transform="" /></svg></button>
                            <button disabled={this.previousPageButton} data-action="previous" aria-label="Go to previous page" className="pf-c-button pf-m-plain" type="button" onClick={this.props.handleMoveLeft}><svg fill="currentColor" height="1em" width="1em" viewBox="0 0 256 512" aria-hidden="true" role="img">
                              <path d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z" transform="" /></svg></button>
                            Page: {this.props.pageNumber}
                            <button disabled={this.nextPageButton} data-action="next" aria-label="Go to next page" className="pf-c-button pf-m-plain" type="button" onClick={this.props.handleMoveRight}><svg fill="currentColor" height="1em" width="1em" viewBox="0 0 256 512" aria-hidden="true" role="img">
                              <path d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z" transform="" /></svg></button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div></div>
              </div>
            </nav>
          </LevelItem>
        </Level>
      </Fragment>
    );
  }

  private onToggle = isOpen => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  };

  private dropDownValue = (value) => {
    this.setState({
      isOpen: !this.state.isOpen,
      itemsPerPage: Number(value.substr(0, value.indexOf(" "))),
      selected: value
    }, () => {
      this.props.handlePerPageLimit(Number(value.substr(0, value.indexOf(" "))))
      this.props.handleItemsPerPage(Number(value.substr(0, value.indexOf(" "))))
    });
  };
}

export { Pagination }