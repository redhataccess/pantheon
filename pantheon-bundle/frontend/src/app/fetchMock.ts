if (!self.fetch) {
    self.fetch = (input: RequestInfo, init?: RequestInit | undefined) => {
        const p = new Promise<Response>((resolve, reject) => {
            // just don't do anything. this exists so that our tests don't try and actually resolve fetches,
            // (which is exactly what isomorphic-fetch tried to do while we were importing it...)
            // and calling nothing here allows us to skip all of our fetch.then() and fetch.catch() code.
        })
        return p
    }
}