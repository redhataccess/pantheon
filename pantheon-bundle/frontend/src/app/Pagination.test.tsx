import React from 'react'
import { Pagination } from '@app/Pagination'
import '@app/fetchMock'

import { mount, shallow } from 'enzyme'
import { LevelItem } from '@patternfly/react-core'

describe('Tests for Pagination', () => {

  test('should render Pagination component', () => {
    const view = shallow(<Pagination 
    handleMoveLeft={move}
    handleMoveRight={move}
    handleMoveToFirst={move}
    pageNumber={1}
    nextPageRecordCount={1}
    handlePerPageLimit={move}
    perPageLimit={1}
    showDropdownOptions={true}
    bottom={true}
/>)
    expect(view).toMatchSnapshot()
  })

  it('should render Badge for displaying page number', () => {
    const wrapper = mount(<Pagination 
      handleMoveLeft={move}
      handleMoveRight={move}
      handleMoveToFirst={move}
      pageNumber={1}
      nextPageRecordCount={1}
      handlePerPageLimit={move}
      perPageLimit={1}
      showDropdownOptions={true}
      bottom={true}
    />)
    const pageNumberDisplay = wrapper.find(LevelItem)
    expect(pageNumberDisplay.exists()).toBe(true)
  })  

  function move(){
    return "R"
  }
})
