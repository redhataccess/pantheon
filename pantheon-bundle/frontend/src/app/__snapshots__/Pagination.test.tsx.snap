// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Pagination should render Pagination component 1`] = `
<Fragment>
  <Level
    gutter="md"
  >
    <LevelItem />
    <LevelItem />
    <LevelItem>
      <nav
        aria-label="Countries Pagination"
      >
        <div>
          <div>
            <div
              className="example"
            >
              <div
                className="ws-preview"
              >
                <div
                  className="pf-c-pagination"
                  id="pagination-options-menu-top"
                >
                  <div
                    className="pf-c-dropdown"
                  >
                    <div
                      className="pf-c-options-menu__toggle pf-m-plain pf-m-text"
                    >
                      <Dropdown
                        newPerPagevalue="25 items per page"
                        perPageValue={[Function]}
                      />
                      <button
                        aria-label="Go to first page"
                        className="pf-c-button pf-m-plain"
                        data-action="first"
                        disabled={true}
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          viewBox="0 0 448 512"
                          width="1em"
                        >
                          <path
                            d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                            transform=""
                          />
                        </svg>
                      </button>
                      <button
                        aria-label="Go to previous page"
                        className="pf-c-button pf-m-plain"
                        data-action="previous"
                        disabled={true}
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          viewBox="0 0 256 512"
                          width="1em"
                        >
                          <path
                            d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                            transform=""
                          />
                        </svg>
                      </button>
                      Page: 
                      1
                      <button
                        aria-label="Go to next page"
                        className="pf-c-button pf-m-plain"
                        data-action="next"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          viewBox="0 0 256 512"
                          width="1em"
                        >
                          <path
                            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                            transform=""
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </nav>
    </LevelItem>
  </Level>
</Fragment>
`;
